src/NodeApi/Runtime/NodeEmbeddingRuntimeSettings.cs

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.JavaScript.NodeApi.Runtime;
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using static NodeEmbedding;
+using static NodejsRuntime;
+
+public class NodeEmbeddingRuntimeSettings
+{
+    public int? NodeApiVersion { get; set; }
+    public NodeEmbeddingRuntimeFlags? RuntimeFlags { get; set; }
+    public string[]? Args { get; set; }
+    public string[]? RuntimeArgs { get; set; }
+    public PreloadCallback? OnPreload { get; set; }
+    public string? MainScript { get; set; }
+    public LoadingCallback? OnLoading { get; set; }
+    public LoadedCallback? OnLoaded { get; set; }
+    public IEnumerable<NodeEmbeddingModuleInfo>? Modules { get; set; }
+    public PostTaskCallback? OnPostTask { get; set; }
+    public ConfigureRuntimeCallback? ConfigureRuntime { get; set; }
+
+    public static JSRuntime JSRuntime => NodeEmbedding.JSRuntime;
```

Comment:
It's not a setting, so I'd suggest removing the property or at least making it private.